### ISSUE TITLE
bug: Kubestellar image is not running in OpenShift: permission denied 

### ISSUE THREAD

**Issue by ronenkat** [kubestellar/kubestellar#855]
### Describe the bug

In Openshift the default behaviour is that Openshift assign a UID/GID to a running container from a range that is defined when the project is created.
The current Kubestellar assumes that the container is running with `kubestellar` as the user.
So when the Kubestellar container starts kcp can't write to `kubestellar-logs` and to `.kcp` directories.

### Steps To Reproduce

N/A

### Expected Behavior

To deploy in Openshift either we need to build the container so it can run with any user that is assigned by Openshift.
Alternatively we can allow the container to run as user `kubestellar` (UID `1000`).

Here is what needs to doe in order to run the kubestellar image on Openshift as user `kubestellar` (UID 1000).
1. A service account needs to be defined and used by the pod:
```
apiVersion: v1
kind: ServiceAccount
metadata:
  name: kubestellar
  namespace: kubestellar
```
2. Change the default security-group from restricted to `anyuid` - this allows the pod to run using `UID 1000` which is assigned to `kubestellar`.
3. Need to update the deployment YAML as follows:
Note the addition of (1) securityContext to the template/spec, (2) securityContext to the container, and (3) adding serviceAccountName to the template spec
```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: kubestellar-server
  namespace: kubestellar
spec:
  selector:
    matchLabels:
      app: kubestellar-server
  replicas: 1
  template:
    metadata:
      labels:
        app: kubestellar-server
    spec:
      securityContext:
        fsGroup: 1000
        supplementalGroups: [1000]
      volumes:
      - name: kubestellar-pv
        persistentVolumeClaim:
          claimName: kubestellar-pvc
      containers:
      - name: kubestellar-server
        image: quay.io/kubestellar/kubestellar:latest
        imagePullPolicy: Always # IfNotPresent
        securityContext:
           runAsUser: 1000
           runAsGroup: 1000
        volumeMounts:
        - name: kubestellar-pv
          mountPath: "/.kcp"
          readOnly: false
        ports:
        - name: ks-port
          protocol: TCP
          containerPort: 6443
      serviceAccountName: kubestellar
```

### Additional Context

N/A

**Comment by ronenkat** [comment]
FYI - A solution has been implemented by @francostellari in a WiP branch.

**Comment by francostellari** [comment]
This issue has been solved by PR #863
